app/models/spree/digital.rb

```diff
@@ -2,14 +2,19 @@ module Spree
   class Digital < ActiveRecord::Base
     belongs_to :variant
     has_many :digital_links, dependent: :destroy
+    has_many :drm_records, dependent: :nullify
```

Comment:
One more thing ™

Why do you nullify here? To keep the drm_record intact for the user to download? That won't work because we delete the digital_link anyway, so the user has no possibility to reach that record. So we can destroy the drm_records as well.

In practice, we will never delete any digitals (or products for that matter).